cloud2sql/arrow/writer.py

```diff
@@ -157,60 +176,112 @@ def write_batch_to_file(batch: ArrowBatch) -> ArrowBatch:
             normalize(path, row)
 
     pa_table = pa.Table.from_pylist(batch.rows, batch.schema)
-    if isinstance(batch.writer, Parquet):
-        batch.writer.parquet_writer.write_table(pa_table)
-    elif isinstance(batch.writer, CSV):
-        batch.writer.csv_writer.write_table(pa_table)
+    if isinstance(batch.writer.format, Parquet):
+        batch.writer.format.parquet_writer.write_table(pa_table)
+    elif isinstance(batch.writer.format, CSV):
+        batch.writer.format.csv_writer.write_table(pa_table)
     else:
         raise ValueError(f"Unknown format {batch.writer}")
-    return ArrowBatch(table_name=batch.table_name, rows=[], schema=batch.schema, writer=batch.writer)
+    return ArrowBatch(
+        table_name=batch.table_name, rows=[], schema=batch.schema, writer=batch.writer, destination=batch.destination
+    )
 
 
 def close_writer(batch: ArrowBatch) -> None:
-    if isinstance(batch.writer, Parquet):
-        batch.writer.parquet_writer.close()
-    elif isinstance(batch.writer, CSV):
-        batch.writer.csv_writer.close()
+    def uploadToS3(path: Path, bucket_uri: str, region: str) -> None:
+        bucket_name: str
+        if bucket_uri.startswith("s3://"):
```

Comment:
Even AWS describes multiple ways of accessing a bucket: https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html

should we support those?

Reply:
good idea. let me add it